src/main.cpp

```diff
@@ -853,23 +854,19 @@ bool AcceptToMemoryPoolWorker(CTxMemPool &pool,
         if (fLimitFree && nFees < ::minRelayTxFee.GetFee(nSize))
         {
             static double dFreeCount = 0;
+            static int64_t nLastFreeTime;
+            static int64_t nFreeLimit = GetArg("-limitfreerelay", 15);
+            static CCriticalSection csLimiter;
+            LOCK(csLimiter);
 
-            // Use an exponentially decaying ~10-minute window:
-            dFreeCount *= std::pow(1.0 - 1.0 / 600.0, (double)(nNow - nLastTime));
-            nLastTime = nNow;
-
-            // -limitfreerelay unit is thousand-bytes-per-minute
             // At default rate it would take over a month to fill 1GB
-            LOG(MEMPOOL, "Rate limit dFreeCount: %g => %g\n", dFreeCount, dFreeCount + nSize);
-            if ((dFreeCount + nSize) >= (nFreeLimit * 10 * 1000 * nLargestBlockSeen / BLOCKSTREAM_CORE_MAX_BLOCK_SIZE))
+            if (RateLimitExceeded(dFreeCount, nLastFreeTime, nFreeLimit, nSize))
```

Comment:
Agreed on the fact that you are using `RateLimitExceeded` but I would rather do it on a new PR. not strong opinion thou.

Reply:
I just want to keep existing behaviour here. This is missing `nLargestBlockSeen` though. I'll push an update.